### PR TITLE
Add a resource for group attribute mapping

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
Adding a resource for group attribute mapping feature.

Requires https://github.com/grafana/grafana-openapi-client-go/pull/103 to be merged first, so that we can bump the openAPI client dependency.

Here is an example of how the new resource can be used to map a role to a group:
```
resource "grafana_role" "report_admin_role" {
  name                   = "Report Administrator"
  uid                    = "report_admin_role_uid"
  auto_increment_version = true
  permissions {
    action = "reports:create"
  }
  permissions {
    action = "reports:read"
    scope  = "reports:*"
  }
}

resource "grafana_group_attribute_mapping" "report_admin_mapping" {
  group_id  = "business_dev_group_id"
  role_uids = [grafana_role.report_admin_role.uid]
}
```

### CODE REVIEW

File: internal/resources/grafana/common_check_exists_test.go [hunk 1]
```diff
@@ -42,14 +42,14 @@ var (
 		func(t *models.NotificationTemplate) string { return t.Name },
 		func(client *goapi.GrafanaHTTPAPI, id string) (*models.NotificationTemplate, error) {
 			resp, err := client.Provisioning.GetTemplate(id)
-			return payloadOrError(resp, err)
+			return payloadOrError[models.NotificationTemplate](resp, err)
```
Comment (IevaVasiljeva) [inline internal/resources/grafana/common_check_exists_test.go]: I couldn't make it to automatically infer types, but I am not sure why. Let me know if you have any ideas.

File: internal/resources/grafana/resource_alerting_message_template.go [hunk 2]
```diff
@@ -145,7 +145,8 @@ func putMessageTemplate(ctx context.Context, data *schema.ResourceData, meta int
 func deleteMessageTemplate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, _, name := OAPIClientFromExistingOrgResource(meta, data.Id())
 
-	_, err := client.Provisioning.DeleteTemplate(name)
+	params := provisioning.NewDeleteTemplateParams().WithName(name)
+	_, err := client.Provisioning.DeleteTemplate(params)
```
Comment (IevaVasiljeva) [inline internal/resources/grafana/resource_alerting_message_template.go]: Looks like the openAPI spec has changed slightly for alerting.

File: internal/resources/grafana/resource_group_attribute_mapping.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+		resourceGroupAttributeMappingID,
+		&resourceGroupAttributeMapping{},
+	).WithLister(listerFunctionOrgResource(listGroupAttributeMappings))
+}
+
+type resourceGroupAttributeMappingModel struct {
+	ID       types.String   `tfsdk:"id"`
+	OrgID    types.String   `tfsdk:"org_id"`
+	GroupID  types.String   `tfsdk:"group_id"`
+	RoleUIDs []types.String `tfsdk:"role_uids"`
```
Comment (spinillos) [inline internal/resources/grafana/resource_group_attribute_mapping.go]: I think that it should be `types.Set` instead. I don't find `[]types.String` any other place.

**Comment by IevaVasiljeva** [comment]
Waiting for https://github.com/grafana/terraform-provider-grafana/pull/1905 to be merged (it updates the openAPI client dependency). Will then rebase on top of that and fix any remaining issues.

**Comment by IevaVasiljeva** [comment]
Closing the PR for now. We can reopen it later if we chose to roll the GAS feature out.

**Pull request closed without merging**